services/hanke-service/src/main/kotlin/fi/hel/haitaton/hello/db/DbController.kt

```diff
@@ -0,0 +1,49 @@
+package fi.hel.haitaton.hello.db
+
+import mu.KotlinLogging
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import java.sql.Connection
+import java.sql.DriverManager
+import java.util.ArrayList
+
+@Component
+@RestController
+@RequestMapping("/api")
+
+class DbController {
+    private val logger = KotlinLogging.logger { }
+
+    @Autowired
+    lateinit var dbConfigProperties: DbConfigProperties
+
+    @GetMapping("/tablenames")
+    fun getAllTables(): String {
+
+        val dbUrl: String = dbConfigProperties.appDatasourceUrl + "?" +
+            "user=" + dbConfigProperties.appDatasourceUsername +
+            "&password=" + dbConfigProperties.appDatasourcePassword
+
+        val con = DriverManager.getConnection(dbUrl)
+
+        logger.info {
+            "Connected to database: " +
+                "${dbConfigProperties.appDatasourceUrl}"
+        }
+        return listTables(con).toString()
+    }
+
+    fun listTables(connection: Connection): ArrayList<String> {
+        var listOfTableNames: ArrayList<String> = arrayListOf()
+        val sql = "select * from pg_tables where schemaname='public';"
+        val response = connection.createStatement().executeQuery(sql)
+        while (response.next()) {
+            listOfTableNames.add(response.getString("tablename"))
+        }
+        logger.info { "query response:" + "$listOfTableNames" }
+        return listOfTableNames
```

Comment:
Usually database connections need to be closed, one way or another. If reusing a connection (which I don't think is happening), at least the resultset and statement should be closed. And those within finally-block.

Reply:
Ah, you are right, thanks for commenting, I will fixed this to close as well!